clippy_lints/src/methods/unnecessary_fold.rs

```diff
@@ -80,7 +93,7 @@ fn check_fold_with_op(
         let mut applicability = Applicability::MachineApplicable;
 
         let turbofish = if replacement.has_generic_return {
-            format!("::<{}>", cx.typeck_results().expr_ty_adjusted(right_expr).peel_refs())
+            format!("::<{}>", cx.typeck_results().expr_ty(expr))
```

Comment:
Does this change make any difference? More specifically was the original code unnecessary